src/lib.rs

```diff
@@ -204,14 +208,15 @@ impl Error {
 impl std::error::Error for Error {
     fn description(&self) -> &str {
         unsafe {
-            let s = zmq_sys::zmq_strerror(*self as c_int) as *const i8;
+            let s = zmq_sys::zmq_strerror(self.clone() as c_int) as *const i8;
+            // NOTE: from_c_str is deprecated
             std::str::from_c_str(s)
```

Comment:
I lost the fight with the borrow checker here, so it's still using the deprecated method instead of the newer one that I can't seem to annotate correctly